protocol/pubsub/v2/protocol_test.go

```diff
@@ -73,3 +73,44 @@ func TestPublishMessageHasOrderingKey(t *testing.T) {
 	err = prot.Send(WithOrderingKey(ctx, orderingKey), test.FullMessage())
 	require.NoError(err)
 }
+
+func attributesInterceptor(attributes map[string]string) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		if method == "/google.pubsub.v1.Publisher/Publish" {
+			pr, _ := req.(*pb.PublishRequest)
+			for _, m := range pr.Messages {
```

Comment:
Should we check to make sure the count of the incoming attributes matches the count of the expected attributes to make sure we don't also have extra ones added by mistake?

Reply:
@andreas-lindfalk ping

Reply:
We have choosen not to use cloudevents and will not ”need this” anymore so do what you want with this pr so to say :)

Reply:
ok thanks for the update